packages/story-editor/src/components/panels/design/videoOptions/videoOptions.js

```diff
@@ -92,25 +92,16 @@ function VideoOptionsPanel({ selectedElements, pushUpdate }) {
 
   const shouldDisplayMuteButton = useMemo(() => {
     return (
-      (isTranscodingEnabled &&
-        !local &&
-        !isMuted &&
-        !isTrimming &&
-        !isTranscoding &&
-        !isExternal &&
-        isSingleElement) ||
-      isMuting
+      isTranscodingEnabled &&
+      isSingleElement &&
+      ((!isMuted && canTranscodeResource(resource)) || isMuting)
     );
-  }, [
-    isTranscodingEnabled,
-    local,
-    isMuted,
-    isTrimming,
-    isTranscoding,
-    isExternal,
-    isSingleElement,
-    isMuting,
-  ]);
+  }, [isTranscodingEnabled, isSingleElement, isMuted, resource, isMuting]);
+
+  const shouldDisableMuteButton = useMemo(
```

Comment:
Disable the button, if trimming/transcoding/muting is running.

Reply:
Using `useMemo` is overkill here. For such simple tasks, the overhead of using `useMemo` is actually making things worse than when not using it at all.

So just do `const shouldDisableMuteButton = !canTranscodeResource(resource);`

Reply:
But resource changes, which by I wrapped it.

Reply:
Even if the resource changes, doing this `canTranscodeResource()` calculation multiple times is most likely still faster than setting up `useMemo`

`useMemo` should only be used for expensive calculations because it adds extra overhead.

https://canimerge.com/should-you-really-use-usememo-in-react-lets-find-out/ is one article that explains this.